gce/update.md

```diff
@@ -0,0 +1,103 @@
+Instructions to Update Pangeo Deployment
+========================================
+
+This document includes detailed instructions on how to update the default
+environment for the specific pangeo.pydata.org deployment.  It may also be of
+general use for other deployments.
+
+
+Update Docker Images
+--------------------
+
+Modify the Dockerfiles in `gce/notebook` and `gce/worker` to suit your needs,
+possibly including new versions of libraries.  Consider also adding new
+examples to the `gce/notebook/examples` directory.
+
+We recommend that you first modify the worker docker image because you will
+need to include its new tag in the `gce/notebook/worker-template.yaml` file.
+
+### Build worker image
+
+Navigate to the worker directory and build the docker image.  We currently use
+the date like `YYYY-MM-DD` as a tag.
+
+```bash
+cd gce/worker
+# edit Dockerfile
+docker build -t daskdev/pangeo-worker:TAG .
+```
+
+You will probably have to do this a few times.
+
+
+### Build Notebook image
+
+Navigate to the `gce/notebook` directory and modify the Dockerfile, notebooks,
+and `worker-template.yaml` file as necessary.  In particular you should update
+the `image:` tag to match the tag of the worker that you have just built.
+
+```bash
+docker build -t daskdev/pangeo-notebook:TAG .
+```
+
+### Push to dockerhub
+
+You can push both images to docker hub:
+
+```bash
+docker push dsakdev/pangeo-worker:TAG
+docker push dsakdev/pangeo-notebook:TAG
+```
```

Comment:
It sounds like it is possible to update the docker images without ever syncing up the pangeo repo. My worry is that, if multiple people start updating the deployment this way, we will break something and not be able to roll back because we don't have the configuration under version control.

A better scenario would be that the docker files and example notebooks are updated via PR, and the rest of the build / deploy process occurs automatically via some sort of CI magic. I recognize that this may be difficult to set up and therefore not practical right now, but I just wanted to raise the possibility.

If we don't go that route, maybe we can update these instructions to remind whoever is doing this to also push the changes to github.

Reply:
I agree that this would be a good way to manage things.  I'm happy to help advise others if they take this on.